### PR TITLE
chore: upgrade psutil

### PR DESCRIPTION
**Pull request by kohlisid**
Kindly explain what this PR does.

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -219,63 +219,63 @@ files = [
...
-    {file = "coverage-7.5.3.tar.gz", hash = "sha256:04aefca5190d1dc7a53a4c1a5a7f8568811306d7a8ee231c42fb69215571944f"},
+    {file = "coverage-7.5.4-cp310-cp310-macosx_10_9_x86_64.whl", hash = "sha256:6cfb5a4f556bb51aba274588200a46e4dd6b505fb1a5f8c5ae408222eb416f99"},
+    {file = "coverage-7.5.4-cp310-cp310-macosx_11_0_arm64.whl", hash = "sha256:2174e7c23e0a454ffe12267a10732c273243b4f2d50d07544a91198f05c48f47"},
+    {file = "coverage-7.5.4-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:2214ee920787d85db1b6a0bd9da5f8503ccc8fcd5814d90796c2f2493a2f4d2e"},
+    {file = "coverage-7.5.4-cp310-cp310-manylinux_2_5_i686.manylinux1_i686.manylinux_2_17_i686.manylinux2014_i686.whl", hash = "sha256:1137f46adb28e3813dec8c01fefadcb8c614f33576f672962e323b5128d9a68d"},
+    {file = "coverage-7.5.4-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl", hash = "sha256:b385d49609f8e9efc885790a5a0e89f2e3ae042cdf12958b6034cc442de428d3"},
+    {file = "coverage-7.5.4-cp310-cp310-musllinux_1_2_aarch64.whl", hash = "sha256:b4a474f799456e0eb46d78ab07303286a84a3140e9700b9e154cfebc8f527016"},
```
Comment (yhl25) [inline poetry.lock:232]: is it a good practice to check in the `poetry.lock` file?

File: pyproject.toml [hunk 2]
```diff
@@ -29,7 +29,7 @@ google-api-core = "^2.11.0"
 protobuf = ">=3.20,<5.0"
 aiorun = "^2023.7"
 uvloop = "^0.19.0"
-psutil = "^5.9.8"
+psutil = ">=5.9.8"
```
Comment (ab93) [inline pyproject.toml:32]:
```suggestion
psutil = "^6.0"
```
But where are we using this in code?

File: poetry.lock (hunk 1 above)
Comment (ab93) [reply poetry.lock:232]: Not needed but doing this helps in reproducible local builds

File: pyproject.toml (hunk 2 above)
Comment (kohlisid) [reply pyproject.toml:32]: https://github.com/numaproj/numaflow-python/blob/1f839005290eaa0aef870c6973d8d16a14223977/pynumaflow/shared/server.py#L284

**Review by ab93 — APPROVED** [review]

**Pull request merged**